src/compact-denylist-format.md

```diff
@@ -0,0 +1,498 @@
+Path and CID blocks respectively).
+
+This specification corresponds to V1 of the compact list format. We have
+limited the number of features and extensions to a minimum to start working
+with, leaving some ideas on the table and the door open to develop the format
+in future versions.
+
+## Specification
+
+### Denylist file extension, locations and order
+
+While not pertaining to the denylist format itself, we introduce the following conventions about denylist files when they are stored in the local filesystem:
+
+- Denylist files MUST be named with the extension `.deny`.
+- Implementations SHOULD look in `/etc/ipfs/denylists/` and
+  `$XDG_CONFIG_HOME/ipfs/denylists/` (default: `~/.config/ipfs/denylists`) for denylist files.
+- Denylist files are processed in alphabetical order so that rules from later
+  denylists override rules from earlier denylists on conflict.
+
+### Denylist format
+
+#### Summary
+
+The following example showcases the features and syntax of a compact denylist:
+
+```yaml
+version: 1
+name: IPFSCorp blocking list
+description: A collection of bad things we have found in the universe
+author: abuse-ipfscorp@example.com
+hints:
+  hint: value
+  hint2: value2
+---
+# Blocking by CID is codec-agnostic (blocks by multihash).
+# Does not block subpaths per se, but might stop an implementation
+# from resolving subpaths if this block is not retrievable.
+/ipfs/bafybeihvvulpp4evxj7x7armbqcyg6uezzuig6jp3lktpbovlqfkuqeuoq
+
+# Blocking by subpath (equivalent rules)
+/ipfs/Qmah2YDTfrox4watLCr3YgKyBwvjq8FJZEFdWY6WtJ3Xt2/test*
+/ipfs/QmTuvSQbEDR3sarFAN9kAeXBpiBCyYYNxdxciazBba11eC/test/*
+
+# Block some subpaths with exceptions: last-matching-rule wins (!)
+/ipfs/QmUboz9UsQBDeS6Tug1U8jgoFkgYxyYood9NDyVURAY9pK/blocked*
+!/ipfs/QmUboz9UsQBDeS6Tug1U8jgoFkgYxyYood9NDyVURAY9pK/blockednot
+!/ipfs/QmUboz9UsQBDeS6Tug1U8jgoFkgYxyYood9NDyVURAY9pK/blocked/not
+!/ipfs/QmUboz9UsQBDeS6Tug1U8jgoFkgYxyYood9NDyVURAY9pK/blocked/exceptions*
+
+# Block IPNS domain name
+/ipns/domain.example
+
+# Block IPNS domain name and path
+/ipns/domain2.example/path
+
+# Block IPNS key - blocks wrapped multihash.
+/ipns/k51qzi5uqu5dhmzyv3zac033i7rl9hkgczxyl81lwoukda2htteop7d3x0y1mf
+
+# Double-hash CID block
+# base58btc-sha256-multihash(QmVTF1yEejXd9iMgoRTFDxBv7HAz9kuZcQNBzHrceuK9HR)
+# Blocks bafybeidjwik6im54nrpfg7osdvmx7zojl5oaxqel5cmsz46iuelwf5acja
+# and QmVTF1yEejXd9iMgoRTFDxBv7HAz9kuZcQNBzHrceuK9HR etc. by multihash
+//QmX9dhRcQcKUw3Ws8485T5a9dtjrSCQaUAHnG4iK9i4ceM
+
+# Double-hash Path block using blake3 hashing
+# base58btc-blake3-multihash(gW7Nhu4HrfDtphEivm3Z9NNE7gpdh5Tga8g6JNZc1S8E47/path)
+# Blocks /ipfs/bafyb4ieqht3b2rssdmc7sjv2cy2gfdilxkfh7623nvndziyqnawkmo266a/path
+# /ipfs/bafyb4ieqht3b2rssdmc7sjv2cy2gfdilxkfh7623nvndziyqnawkmo266a/path
+# /ipfs/f01701e20903cf61d46521b05f926ba1634628d0bba8a7ffb5b6d5a3ca310682ca63b5ef0/path etc...
+# But not /path2
+//QmbK7LDv5NNBvYQzNfm2eED17SNLt1yNMapcUhSuNLgkqz
```

Comment:
TODO: we need to update this example after https://github.com/ipfs-shipyard/nopfs/pull/28/files lands, and have double-hash entry that is Multihash that is something other than sha256 (e.g. `gW813G35CnLsy7gRYYHuf63hrz71U1xoLFDVeV7actx6oX` from the linked PR)